File: foody/src/main/java/com/example/foody/service/SubscribeService.java
package com.example.foody.service;

import com.example.foody.model.Subscribe;
import com.example.foody.repository.SubscribeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public class SubscribeService {

    @Autowired
    private SubscribeRepository repository;

    // Save new email
    public Subscribe saveEmail(String email) {
        if (repository.existsByEmail(email)) {
            return null;
        }

        Subscribe subscriber = new Subscribe();
        subscriber.setEmail(email);

        return repository.save(subscriber);
    }

    // Get all subscribers with pagination
    public Page<Subscribe> getAllSubscribers(Pageable pageable) {
        return repository.findAll(pageable);
    }

    // Search subscribers by email (with pagination)
    public Page<Subscribe> searchByEmail(String keyword, Pageable pageable) {
        return repository.findByEmailContainingIgnoreCase(keyword, pageable);
    }

    public void deleteSubscriber(Long id) {
       repository.deleteById(id);
    }
}

File: FoodyUI/src/Service/SubscribeService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/subscribe';

// Subscribe using email
export const subscribeWithEmail = (email) => {
  return axios.post(API_BASE_URL, null, { params: { email } });
};

// Get all subscribers with pagination
// Example: getAllSubscribers(0, 10)
export const getAllSubscribers = (page = 0, size = 10) => {
  return axios.get(`${API_BASE_URL}/all`, {
    params: { page, size }
  });
};

// Search subscribers by email keyword with pagination
// Example: searchSubscribers("john", 0, 10)
export const searchSubscribers = (keyword, page = 0, size = 10) => {
  return axios.get(`${API_BASE_URL}/search`, {
    params: { keyword, page, size }
  });
};

// Delete subscriber by ID
export const deleteSubscriber = (id) => {
  return axios.delete(`${API_BASE_URL}/${id}`);
};

export const getSubscriberCount = () => {
    return axios.get(`${API_BASE_URL}/count`);
}
